### PR TITLE
Sensor Library - OPT2002

### PR DESCRIPTION
**Pull request by JLefebvre55**
Test Script Author: Alex Zeng @AlexZengXi 
Current Hardware Owner: Jayden Lefebvre @JLefebvre55 
Final Review: Urmil Modi @urmilmodi 

1. Author reviews library contents
2. Jayden Lefebvre writes Arduino test script using library
3. Current owner tests library implementation on hardware, documents results

Comment on this PR if you have any questions

### CODE REVIEW

File: OPT2002/src/OPT2002.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    // For Arduino Mega use a 200 ohm resistors, 10V input read as 5.0V at the Arduino
+    
+    // Calibrate from analogRead to 10V (undoing the resistor)
+    double voltage = opt2002_map(analogRead(inpin), 0, 1023, 0, 10);
+    double current = voltage / resistor;
+    double distance = 0
+
+    if(current != 0){
+        // Calibrate from 30mm to 80mm
+        double distance = opt2002_map(current, 4, 20, 30, 80);
```
Comment (urmilmodi) [inline OPT2002/src/OPT2002.cpp]: The calibration here is different than the test script one, is there a reason why? We should be using the calibration each sensor requires.

**Review by urmilmodi — CHANGES_REQUESTED** [review]
Everything seems fine, except the calibration changes.

File: OPT2002/src/OPT2002.cpp (hunk 1 above)
Comment (JLefebvre55) [reply OPT2002/src/OPT2002.cpp]: @urmilmodi From the commit history, seems like @AlexZengXi made some changes on those lines after I created the libary. Alex - is this the right code?

**Comment by JLefebvre55** [comment]
Fixed today with @AlexZengXi, confirmed functionality. Small bug with distances between 60 and 80 mm - we believe this to be a limited current output problem. @AlexZengXi will figure out if there's a current output maximum and will redesign the voltage divider to account for that. Haven't tested error reporting.

**Review by AlexZengXi — APPROVED** [review]

**Comment by AlexZengXi** [comment]
Testing documentation:

Screenshot (133)

Tested with ruler
Produced results +/- 3mm accuracy

**Pull request merged**